File: ts/src/library/Json.ts
interface Point {
    x: number;
    y: number;
}

export function run(): void {
    let point: Point = { x: 1.0, y: 2.0};
    let serialized = JSON.stringify(point);
    console.log(`serialized = ${serialized}`);
    let deserialized = JSON.parse(serialized) as Point;
    console.log(`deserialized = Point(${deserialized.x}, ${deserialized.y})`);
}

File: rust/src/library/json.rs
/**
  * Created by sebby on 9/25/16.
  */

use serde_json;

#[derive(Serialize, Deserialize, Debug)]
struct Point {
    x: f32,
    y: f32,
}

pub fn run() {
    let point = Point { x: 1.0, y: 2.0 };
    let serialized = serde_json::to_string(&point).unwrap();
    println!("serialized = {}", serialized);
    let deserialized: Point = serde_json::from_str(&serialized).unwrap();
    println!("deserialized = {:?}", deserialized);
}
